esx_service/utils/auth_data.py

```diff
@@ -576,7 +638,9 @@ def connect(self):
 
         if not need_create_table:
             if os.path.isfile(self.db_path) and self.need_upgrade_db():
-                self.handle_upgrade_db()
+                error = self.handle_upgrade_db()
+                if error:
```

Comment:
Not this change, but shouldn;t the DB upgrade check happen once on service startup and service falls back to existing version if upgrade fails or service refuses to start which is catastrophic - data loss (no volumes can be used).

Believe this change is really not so critical to prevent using data volumes for an upgrade failure and auth_* code should be able to simply fall back to using VM name from UUID if available or no-name. auth_* code should be able to work with a 1.1 DB also if required. Simply, service can't prevent using user data!